test-case/check_xrun_injection.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+##
+## Case Name: check-xrun-injection
+## Preconditions:
+##    N/A
+## Description:
+##    check xrun injection during playback/capture
+##    default duration is 10s
+##    default interval time of xrun injection is 0.5s
+## Case step:
+##    1. Parse TPLG file to get pipeline
+##    2. Specify the audio parameters
+##    3. Run aplay or arecord on each pipeline with parameters
+##    4. do xrun injection during playback or capture
+## Expect result:
+##    The return value of aplay/arecord should be 0
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $TPLG'
+OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
+
+OPT_OPT_lst['d']='duration' OPT_DESC_lst['d']='aplay/arecord duration in second'
+OPT_PARM_lst['d']=1         OPT_VALUE_lst['d']=10
+
+OPT_OPT_lst['i']='interval'     OPT_DESC_lst['i']='interval time of xrun injection'
+OPT_PARM_lst['i']=1         OPT_VALUE_lst['i']=0.5
+
+OPT_OPT_lst['f']='file'   OPT_DESC_lst['f']='source file path'
+OPT_PARM_lst['f']=1         OPT_VALUE_lst['f']=''
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+
+tplg=${OPT_VALUE_lst['t']}
+duration=${OPT_VALUE_lst['d']}
+interval=${OPT_VALUE_lst['i']}
+file=${OPT_VALUE_lst['f']}
+
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+# checking if source file exists
+if [[ -z $file ]]; then
+    dlogi "no source file, use /dev/zero as dummy playback source"
+    file=/dev/zero
+elif [[ ! -f $file ]]; then
+    dlogw "$file does not exist, use /dev/zero as dummy playback source"
+    file=/dev/zero
+else
+    dlogi "using $file as playback source"
+fi
+
+func_lib_setup_kernel_last_line
+func_lib_check_sudo
+func_pipeline_export $tplg "type:playback,capture,both" 
+
+xrun_injection()
+{
+    count=1
+    while(true)
+    do
+    	ps -ef |grep "$pid"| grep -v grep
```

Comment:
1.  `$pid` is parameter, here is not clear
2.  you can use `pidof` to confirm target command still in the process, you also can use sof-process-state do the more check

Reply:
I think "pid=$!" is safer,  pidof will list all the pid num of cmdlines. Suppose there was more than one aplay process，the pid num will not be unique.
In this part, I only want to check if target pid is still running or not.  The return value will be checked by wait "$pid".